### PR TITLE
Bump signac and signac-flow

### PR DESCRIPTION
**Pull request by tommy-waltmann**
## Description

This PR bumps both `signac` and `signac-flow`, taking the place of #47 and #48 . It also fixes a typo in `requirements.txt`

## Motivation and context

Bumping `signac` and `signac-flow` individually will necessary cause errors, due to the major version bump in `signac` and the new version requirements established between the packages.

## How has this been tested?

If CI tests pass, we are good to go.

## Checklist:

- [ ] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-validation/blob/trunk/CONTRIBUTING.md).
- [ ] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-validation/blob/trunk/ContributorAgreement.md).
- [ ] My name is on the list of contributors (`CONTRIBUTORS.md`) in the pull request source branch.

### CODE REVIEW

File: .github/workflows/requirements-test.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-signac==1.8.0
-signac-flow==0.24.0
+signac==2.0.0
+signac-flow==0.25.1
```
Comment (joaander) [inline .github/workflows/requirements-test.txt:2]:
```suggestion
signac-flow==0.25.0
```
0.25.1 doesn't exist yet: https://pypi.org/project/signac-flow/#history

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
-signac >= 1.8.0
-signac-flow >= 1.22.0
+signac >= 2.0.0
+signac-flow >= 0.25.1
```
Comment (joaander) [inline requirements.txt:2]:
```suggestion
signac-flow >= 0.25.0
```

**Review by joaander — CHANGES_REQUESTED** [review]

**Comment by tommy-waltmann** [comment]
It appears that they merged a [release branch](https://github.com/glotzerlab/signac-flow/pull/742) a few days ago, but the release doesn't seem to have been made yet. The 0.25.0 release of flow had a critical bug where the -f flag couldn't be used on the command line ([see here](https://github.com/glotzerlab/signac-flow/pull/738)), so we should wait until the 0.25.1 release is made before merging this.

**Comment by joaander** [comment]
This bug prevents all use of `signac-flow`? Or just the `-f` command line option? `hoomd-validation` does not rely on the `-f` option.

**Comment by tommy-waltmann** [comment]
Looks like 0.25.1 is on PyPI now: https://pypi.org/project/signac-flow/

The conda-forge package should be available later today or tomorrow

**Comment by joaander** [comment]
`hoomd-validation` uses PyPI.

**Comment by joaander** [comment]
I reran the build. It does appear that we are using removed APIs that need updating.
```
    project = signac.init_project(root=config.project_root)
TypeError: init_project() got an unexpected keyword argument 'root'
```

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**